vmdb/lib/workers/event_catcher_kubernetes.rb

```diff
@@ -37,9 +37,38 @@ def monitor_events
     reset_event_monitor_handle
   end
 
+  EVENTS_FILTER = Hash.new { [] }.merge!(
+    'Node' => %w(NodeReady NodeNotReady rebooted),
+    'Pod'  => %w(scheduled)
+  )
+
   def process_event(event)
-    # TODO: some sane event filtering
-    $log.info "#{log_prefix} Caught event [#{event}]"
-    # TODO: add_queue kubernetes event
+    event_data = {
+      :timestamp => event.object.lastTimestamp,
+      :kind      => event.object.involvedObject.kind,
+      :name      => event.object.involvedObject.name,
+      :namespace => event.object.involvedObject['table'][:namespace],
+      :reason    => event.object.reason,
+      :message   => event.object.message
+    }
+
+    supported_reasons = EVENTS_FILTER[event_data[:kind]]
+
+    unless supported_reasons.include?(event_data[:reason])
+      $log.debug "#{log_prefix} Discarding event [#{event_data}]"
```

Comment:
This type of event filtering is usually done by adding the event types to the [`filtered_events` list in the `event_handling` file](https://github.com/ManageIQ/manageiq/blob/e3cc148b379a9da7f1676fbdb40de0b5b5896e62/vmdb/config/event_handling.tmpl.yml#L1407).  This allows us to control the processed events via configuration instead of code changes.

However, it might make sense to short circuit that here like you're doing if the total number of events is expected to be extremely high.

Since you have an "inclusion" list instead of an "exclusion" list, maybe it makes more sense to just add this list to the [`event_catcher_kubernetes` configuration in the `vmdb` config file](https://github.com/ManageIQ/manageiq/blob/e3cc148b379a9da7f1676fbdb40de0b5b5896e62/vmdb/config/vmdb.tmpl.yml#L316).  It would make it easier to enable future events without having to make actual code changes.

Also, I'm OK with this type of change coming later.

Thoughts?